src/components/NewOutgoingComponent.jsx

```diff
@@ -0,0 +1,105 @@
+            outgoingCategory: '',
+            outgoingDate: ''
+        },
+    };
+
+    render() {
+        return (
+            <FormControl onSubmit={this.addOutgoing}>
+                {this.state.isOutgoingSaved && <Redirect to='/outgoings' push/>}
+                <TextField
+                    id="outgoing-title" name="outgoingTitle" type="text" placeholder="Titel eingeben"
+                    autoComplete="on"
+                    value={this.state.outgoing.outgoingTitle}
+                    onChange={(event) => {
+                        this.setState({outgoing: { ...this.state.outgoing, outgoingTitle: event.target.value}})
+                    }}
+                />
+                <FormControl>
+                    <InputLabel htmlFor="amount">Betrag</InputLabel>
+                    <Input
+                        id="amount"
+                        type="number"
+                        value={this.state.outgoing.outgoingAmount}
+                        onChange={(event) => {
+                            this.setState({outgoing: { ...this.state.outgoing, outgoingAmount: event.target.value}})
+                        }}
+                        startAdornment={
+                            <InputAdornment position="start">CHF</InputAdornment>
+                        }
+                    />
+                </FormControl>
+                <TextField
+                    id="outgoing-categorie" name="outgoingCategorie" type="text" placeholder="Kategorie eingeben"
+                    autoComplete="on"
+                    value={this.state.outgoing.outgoingCategory}
+                    onChange={(event) => {
+                        this.setState({outgoing: { ...this.state.outgoing, outgoingCategory: event.target.value}})
+                    }}
+                />
+                <TextField
+                    id="outgoing-categorie" name="outgoingCategorie" placeholder="Datum auswählen"
+                    autoComplete="on"
+                    type="date"
+                    value={this.state.outgoing.outgoingDate}
+                    onChange={(event) => {
+                        this.setState({outgoing: { ...this.state.outgoing, outgoingDate: event.target.value}})
+                    }}
+                />
+                <IconButton
+                    aria-label="add outgoing"
+                    onClick={this.addOutgoing}
+                >
+                    <AddIcon/>
+                </IconButton>
+            </FormControl>
+
+        );
+    }
+
+    addOutgoing = () => {
+        try {
+            this.props.dispatch(doAddOutgoing(this.state.outgoing));
+            this.setState({isOutgoingSaved: true, outgoing: {
+                    outgoingTitle: '',
+                    outgoingAmount: '',
+                    outgoingCategory: '',
+                    outgoingDate: ''
+                }});
+        } catch (e) {
+            console.log(e);
+        }
+    }
+}
+
+const uuid = ()=> {
```

Comment:
Wenn wir UUIDs verwenden, würde ich diese Funktion in eine Helper-Datei auslagern.